### PR TITLE
Some small edits to help clarify parameter validation guidance

### PR DESCRIPTION
**Pull request by annelo-msft**
This is driven by updates to how we validate parameters in the .NET data plane code generator.

### CODE REVIEW

File: docs/dotnet/introduction.md [hunk 1]
```diff
@@ -522,7 +522,7 @@ If in common scenarios, users are likely to pass just a small subset of what the
 
 ##### Parameter Validation
 
-Service methods take two kinds of parameters: _service parameters_ and _client parameters_. _Service parameters_ are directly passed across the wire to the service.  _Client parameters_ are used within the client library and aren't passed directly to the service.
+Service methods take two kinds of parameters: _service parameters_ and _client parameters_. _Service parameters_ are directly passed across the wire to the service.  _Client parameters_ are used within the client library to create the request that will be sent to the service.
```
Comment (KrzysztofCwalina) [inline docs/dotnet/introduction.md:525]: I have to say it does not clarify the guidance for me. There are no parameters passed to service methods that are literally "send directly". All of them are used to "create the request".

File: docs/general/implementation.md [hunk 2]
```diff
@@ -57,7 +57,7 @@ where _ServiceName_ is the canonical shortname without spaces, and _Configuratio
 
 ## Parameter validation
 
-The service client will have several methods that perform requests on the service.  _Service parameters_ are directly passed across the wire to an Azure service.  _Client parameters_ are not passed directly to the service, but used within the client library to fulfill the request.  Examples of client parameters include values that are used to construct a URI, or a file that needs to be uploaded to storage.
+The service client will have methods that send requests to the service.  These methods can take two types of parameters: _client parameters_ and _service parameters_.  _Service parameters_ are passed across the wire to an Azure service.  _Client parameters_ are not passed directly to the service, but used within the client library to create the request.  Examples of client parameters include values that are used to construct a URI (e.g. path parameters), or the path to file that needs to be opened by the client to create the upload request to send to storage.
```
Comment (KrzysztofCwalina) [inline docs/general/implementation.md:60]: I think the reason we validate path parameters is not that they are used to "create the request" (construct the URI), but rather that bad path parameters might cause us to call a different endpoint from what we except and the endpoint might have unexpected and possibly dangerous behavior.

File: docs/dotnet/introduction.md (hunk 1 above)
Comment (annelo-msft) [reply docs/dotnet/introduction.md:525]: That's fair - do you have any suggestions for this that you'd like to see in the .NET guidance?

File: docs/general/implementation.md (hunk 2 above)
Comment (annelo-msft) [reply docs/general/implementation.md:60]: I'll update it to reflect this - thanks!

File: docs/dotnet/introduction.md (hunk 1 above)
Comment (JeffreyRichter) [reply docs/dotnet/introduction.md]: Service methods take two kinds of parameters: _service parameters_ and _client parameters_. _Service parameters_ are sent across the wire to the service as URL segments, query parameters, request header values, & request bodies (typically JSON or XML).  _Client parameters_ are used solely within the client library and are not sent to the service; examples are CancellationTokens or file paths).

File: docs/general/implementation.md (hunk 2 above)
Comment (JeffreyRichter) [reply docs/general/implementation.md]: Yes, what Krzysztof says is correct.
Server parameters used to construct the URL and any mandatory parameters should be null-checked; other server parameters should not be null-checked.

**Review by JonathanGiles — APPROVED** [review]

**Pull request merged**